pilot/pkg/serviceregistry/kube/controller/endpoints.go

```diff
@@ -208,6 +182,42 @@ func (e *endpointsController) buildIstioEndpoints(endpoint interface{}, host hos
 	return endpoints
 }
 
+func (e *endpointsController) buildServiceInstances(ep *v1.Endpoints, ss v1.EndpointSubset, endpoints []v1.EndpointAddress,
+	svc *model.Service, discoverabilityPolicy model.EndpointDiscoverabilityPolicy, labelsList labels.Collection,
+	svcPort *model.Port, health model.HealthStatus) []*model.ServiceInstance {
```

Comment:
I think I must miss something but I don't think `health` is used in this function?

Reply:
it is used 
istioEndpoint.HealthStatus = health

Reply:
hmm I don't see `health` between line 188 to 219.

Reply:
Oh I see, I must look at an old commit.